lxd/instances_post.go

```diff
@@ -1057,7 +988,7 @@ func instancesPost(d *Daemon, r *http.Request) response.Response {
 			logger.Debug("No name provided for new instance, using auto-generated name", logger.Ctx{"project": targetProjectName, "instance": req.Name})
 		}
 
-		if clustered && !clusterNotification && targetMemberInfo == nil {
+		if clustered && !clusterNotification {
```

Comment:
OK this is good, now that you've moved the targetMemberInfo definition down to where its first used, its forced us to highlight the changes of behaviour that are inadvertently being made.

So previously we were resolving the `targetMemberInfo` info earlier so that if a specific target was specified we then did not need to go an perform the (pretty expensive by the looks of it) call to `instance.SuitableArchitectures`.

So looks like call to `project.CheckTarget` needs to be moved up so we can keep the check against `targetMemberInfo == nil` here.

What do you think?

Reply:
This made me thinking. It would report error messages that wouldn't have been shown before since the target checks are performed at a later point in time which would definitely break the expected behavior here. 

To have as little changes as necessary, I have moved back the fetching of all cluster members and the target check to its original place, but still keeping the definition of `targetGroupName` as close to its first usage as possible :)

Reply:
I'll resolve this since it is not part of the PR anymore.